modules/common-fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/common/fs2/EnrichUtils.scala

```diff
@@ -0,0 +1,41 @@
+package com.snowplowanalytics.snowplow.enrich.common.fs2
+
+import _root_.io.circe.Json
+import _root_.io.circe.JsonObject
+import _root_.io.circe.syntax._
+import com.snowplowanalytics.snowplow.analytics.scalasdk.ParsingError
+
+object EnrichUtils {
+
+  def isoDateToMicrosSinceEpochOrSelf(v: Json): Json =
+    v.asString match {
+      case Some(mightBeIso) if TimeUtils.isIsoDate(mightBeIso) =>
+        val time = TimeUtils.isoDateToInstant(mightBeIso)
+        val micros = TimeUtils.microSecondsSinceEpoch(time)
+        Json.fromLong(micros)
+      case _ => v
+    }
+
+  def mapIsoDatesToMicros(fields: Map[String, Json]): Map[String, Json] =
+    fields.mapValues(EnrichUtils.isoDateToMicrosSinceEpochOrSelf)
+
+  def transformTsvToBigQueryJson(tsv: String): Either[ParsingError, Json] = {
```

Comment:
I'd receive the Event model instead, same for the flattened-json method, every other method can be package-private